liblangutil/Scanner.cpp

```diff
@@ -271,12 +274,60 @@ bool Scanner::skipWhitespaceExceptUnicodeLinebreak()
 	return sourcePos() != startPosition;
 }
 
+
+/// Tries to scan for an RLO/LRO/RLE/LRE/PDF and keeps track of script writing direction override depth.
+///
+/// @returns ScannerError::NoError in case of successful parsing and directional encodings are paired
+///          and error code in case the input's lexical parser state is invalid and this error should be reported
+///          to the user.
+static ScannerError parseUnicodeDirectionDepth(shared_ptr<CharStream> const& _stream, size_t _startPosition)
```

Comment:
Usually we use an anonymous namespace (can also be just local around this function).

Reply:
Also this function is not about parsing any depth (at least not when seen externally), it is about checking the sanity or well-formed-ness of a string subsequence.

Reply:
Added namespace around function and renamed it to `isUnicodeSubsequenceWellFormed`